### DISCUSSION TITLE
Adding ability for setValue accept array of files in case of working with <input type='file' />

### DISCUSSION THREAD

**Discussion by Arsikod** [react-hook-form/react-hook-form#6419] [Q&A]
**Is your feature request related to a problem? Please describe.**
I have an `<input type='file' multiple/>`. And whenever I upload multiple files it is stored in "readonly" FileList. Whenever I want to delete **one** of the files, and since I cannot modify FileList I need to get rid of FileList by doing Array.from(FileList).
Now I have array of files that I need to send via methods.setValue([my array of files]).
Array of files is what I need in my final data that I am willing to submit to backend.
But at methods.setValue() moment I hit this error
![image](https://user-images.githubusercontent.com/47626872/131607033-12faa1b2-88d8-47f1-8b8a-09f2abf89645.png)
**the problem is that ref tells that input is type='file' and it tries to cast array of files to FileList**
`266 |   _f.ref.files = value as FileList;`

**Describe the solution you'd like**
I guess it will do no harm to change this line of code 
`266 |   _f.ref.files = value as FileList;`

to this:
`266 |   _f.ref.files = value as FileList | File[];`

so it will make it possible to alter initially uploaded group of files.

**Describe alternatives you've considered**
for now as a workaround I created another hidden input and gave it ref
`   <input
          id="file-upload"
          type="file"
          placeholder={placeholder}
          onChange={handleFileUpload}
          name={name}
          multiple={true}
        />  
        <input ref={ref} />`   -----this helps me to avoid getting into line 266 at useForm.ts

**Additional context**
I googled this issue a lot, and seems like devs dealing with attach multiple file and manipulating have tough times with FileList due to its readonly nature.

**Comment by bluebill1049** [comment]
thanks for the feature request. I think this one is a pretty trick, file input has some security constraints.
